### DISCUSSION TITLE
How to left join one a one-to-one relationship?

### DISCUSSION THREAD

**Discussion by thobson** [prisma/prisma#16648] [Q&A] [answered]
Given this schema

```
model Profile {
  id Int @id @default(autoincrement())
  name String
  user User @relation(fields: [userId], references: [id])
  userId Int @unique    
}

model User {
  id Int @id @default(autoincrement())
  username String
  profile Profile?
}
```

How can I fetch a user and profile without executing two selects? This code:

```
const ent = await this.prisma.user.findUnique({ 
  where: { 
    id: id
  },
  include: {
    profile: true
  }
})
```

Results in a select for the user followed by a select for the profile.

I would like to execute a `SELECT u.username, p.name FROM User u LEFT JOIN Profile p on p.userId = u.id` type query.

My real world example is of course more complex, I actually have 5 relations so I want to avoid 5 hits.

Thanks!

**Comment by thobson** [comment, marked as answer]
Sorry, I just came across https://github.com/prisma/prisma/issues/5184

Coming from Java/Hibernate land I just assumed Prisma supported joins :(

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
